hw/hw11/Mailroom-Madness.py

```diff
@@ -0,0 +1,116 @@
+import sys
+
+donorList = ["Bill Gates", "Elon Musk", "Your Mom",
+             "Harry Potter", "Tony Stark"]
+donations = {"Bill Gates": [200, 600, 300], "Elon Musk": [500, 400],
+             "Your Mom": [5, 20], "Harry Potter": [38.24],
+             "Tony Stark": [500, 500, 500]}
+
+print("Welcome to Mailroom Madness")
+
+
+def menu():
+    print("Choose from the following:")
+    print("T - Send a (T)hank You")
+    print("R - Create a (R)eport")
+    print("quit - Quit the program")
+    choice = input()
+    choice = choice.lower()
+    if choice == "t":
+        thankyou()
+    elif choice == "r":
+        report()
+    elif choice == 'quit':
+        sys.exit()
+    else:
+        print("Please select one of the options.")
+        menu()
+
+
+def thankyou():
+    # Generate a thank you message
+    print("Please enter a name, or choose from the following:")
+    print("list - Print a list of previous donors")
+    print("quit - Return to main menu")
+    donor_name = choosename()
+    if donor_name not in donorList:
+        donorList.append(donor_name)
+        donations[donor_name] = [0]
+    new_donation = float(donationAmount())
+    donations[donor_name].append(new_donation)
+    thankyouMessage = ("Dear %s, \n Thank you so much for your " % donor_name +
+                       "kind donation of $%s0. We here at " % new_donation +
+                       "the Foundation for Homeless Whales greatly " +
+                       "appreciate it. Your money will go towards creating " +
+                       " new oceans on the moon for whales to live in. \n " +
+                       "Thanks again, \n Jim Grant \n Director E. H. W.")
+    print(thankyouMessage)
+    print(input("Press Enter to Continue..."))
+    menu()
+
```

Comment:
Here's an alternate way to string format in python. Ultimately there's no right answer here. I prefer this way because it is more readable to me, and the variables being replaced are easily identifiable:

```
thankyou_message2 = ("""\
        Dear {donor_name},\n
        Thank you so much for your kind donation of ${new_donation:.2f}.
        We here at the Foundation for Homeless Whales greatly appreciate it.
        Your money will go towards creating new oceans on the moon for whales
        to live in.\n
        Thanks again,
        Jim Grant
        Director E. H. W.""".format(donor_name=donor_name,
                                    new_donation=new_donation))
```